.github/workflows/check_changelog.yml

```diff
@@ -0,0 +1,29 @@
+name: Check Changelog
+
+on:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'no-changelog') == 0 }}
+    steps:
+      - name: Get PR number and author
+        run: |
+          echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+          echo "PR_AUTHOR=${{ github.event.pull_request.user.login }}" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Check the changelog
+        run: |
+          set -xe
+          result=`grep "$PR_NUMBER" ./doc/whats_new/v*.rst | grep "$PR_AUTHOR"`
```

Comment:
Is this forcing to have the pr number and author on the  same line? This might not be practical.

Reply:
You are right! The two grep are not a good idea. I will check the PR number and if tests are modified then. Thanks.

Reply:
Done (this pr passes the test... :) ): I prefer to keep the "no-changelog" label anyway, just in case we need to manually avoid the check.